source-repo-scripts/prepare_for_release.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2024 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import datetime
+from typing import Optional
+import subprocess
+import re
+from pathlib import Path
+
+REPO_NAMES = {
+    "gz-cmake": "Gazebo CMake",
+    "gz-utils": "Gazebo Utils",
+    "gz-tools": "Gazebo Tools",
+    "gz-math": "Gazebo Math",
+    "gz-plugin": "Gazebo Plugin",
+    "gz-common": "Gazebo Common",
+    "gz-msgs": "Gazebo Msgs",
+    "sdformat": "libsdformat",
+    "gz-fuel-tools": "Gazebo Fuel Tools",
+    "gz-transport": "Gazebo Transport",
+    "gz-physics": "Gazebo Physics",
+    "gz-rendering": "Gazebo Rendering",
+    "gz-sensors": "Gazebo Sensors",
+    "gz-gui": "Gazebo GUI",
+    "gz-sim": "Gazebo Sim",
+    "gz-launch": "Gazebo Launch",
+}
+
+
+def ext_run(cmd: list):
+    # print(f"Running: {' '.join(cmd)}")
+    po = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = po.communicate()
+    if po.returncode != 0:
+        print("Error running command (%s)." % (" ".join(cmd)))
+        print("stdout: %s" % (out.decode()))
+        print("stderr: %s" % (err.decode()))
+        raise Exception("subprocess call failed")
+    return out.decode()
+
+CMAKE_PROJECT_VERSION_PATTERN = r"^project\W*\(\W*([a-z0-9_-]*)\W*VERSION\W*([0-9.]*)"
+
+def match_cmake_project_version(file_content: str):
+    return re.search(
+        CMAKE_PROJECT_VERSION_PATTERN ,
+        file_content,
+        re.MULTILINE,
+    )
+
+def get_project_and_version_from_cmake(cmake_file: str):
+    with open(cmake_file) as f:
+        m = match_cmake_project_version(f.read())
+    if m:
+        return m.groups()
+    else:
+        raise RuntimeError("Could not parse project and version from CMakeLists.txt")
+
+def get_repo_from_project(project: str):
+    repo = project.replace("ignition", "gz").replace("gazebo", "sim")
+    repo = repo.replace('_','-')
+    # remove the version
+    m = re.search("[a-z-_]*", repo)
+    print(m)
+    if not m:
+        raise RuntimeError("Could not determine repo from project name")
+    else:
+        repo = m.group(0)
+        return repo, REPO_NAMES[repo]
+
+
+def calculate_new_version(bump: str, previous_version: str):
+
+    new_version = [int(v) for v in previous_version.split(".")]
+    if bump == "major":
+        new_version[0] += 1
```

Comment:
I think that when bumping the major version we also need to set minor and patch to `0`. Same when bumping the minor version, we need to set patch to `0`.